docs/src/main/tut/auth.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: docs
+title: Authentication API
+---
+
+# Authentication API
+
+Github4s supports the [Authentication API](https://developer.github.com/v3/oauth_authorizations/). As a result,
+with github4s, you can:
+
+- [New authentication](#new-authentication)
+- [Authorize url](#authorize-url)
+- [Get access token](#get-access-token)
+
+The following examples assume the following imports:
+
+```tut:silent
+import github4s.Github
+import github4s.Github._
+import github4s.jvm.Implicits._
+import scalaj.http.HttpResponse
+// if you're using ScalaJS, replace occurrences of HttpResponse by SimpleHttpResponse
+//import github4s.js.Implicits._
+//import fr.hmil.roshttp.response.SimpleHttpResponse
+
+```
+
+They also make use of `cats.Id` but any type container implementing `MonadError[M, Throwable]` will
+do.
+
+Support for `cats.Id`, `cats.Eval` and `Future` (the only supported option for scala-js) are
+provided out of the box when importing `github4s.{js,jvm}.Implicits._`.
+
+## Authentication
```

Comment:
Since I think there is only Auth-related stuff in this API I don't think this level is necessary.
It's not like Repos where you have Branches, Statuses, etc